src/inmanta/deploy/scheduler.py

```diff
@@ -675,37 +700,85 @@ async def report_resource_state(
             if deployment_result is not None:
                 # first update state, then send out events
                 self._deploying_latest.remove(resource)
+
+                previous_deployment_result = state.deployment_result
+
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+
+                # Set of dependant resources that are interested in deploys of the current resource
+                concerned_resources: set[ResourceIdStr] = set()
+                provides: Set[ResourceIdStr] = self._state.requires.provides_view().get(resource, set())
+
                 if deployment_result is DeploymentResult.DEPLOYED:
                     self._state.dirty.discard(resource)
+                    is_dirty = False
+
+                    if previous_deployment_result != DeploymentResult.DEPLOYED:
+                        # This resource went from not deployed to deployed
+                        # we have to inform its dependants regardless of event propagation
+                        # so that we can try to re-deploy them in case this resource's previous
+                        # failure was the reason they were skipped
+
+                        # FIXME: stop informing resource skipped because of
+                        #  a custom handler skip via this mechanism
+
+                        for dependant in provides:
+                            if not self._state.resources.get(dependant, None):
+                                continue
+                            if not (dependant_status := self._state.resource_state.get(dependant, None)):
+                                continue
+                            if dependant_status.blocked is BlockedStatus.NO:
+                                concerned_resources.add(dependant)
+                                # Remove timers for unblocked dependant resources because
+                                # they are marked for deployment below.
+                                self._timer_manager.remove_resource(dependant)
```

Comment:
I wonder if we should remove them here. 

It is purely an optimization: (if it fires it won't change anything if priority is lower, if priority is higher, this remove is wrong.
 
So if we change the priorities, this optimization is a hard to find bug.

Reply:
removed